ops/pebble.py

```diff
@@ -1308,8 +1318,8 @@ class FileInfo:
     name: str
     """Base name of the file."""
 
-    type: Union[FileType, str]
-    """Type of the file ("file", "directory", "symlink", etc)."""
+    type: FileType
+    """Type of the file (FileType.FILE, FileType.DIRECTORY, FileType.SYMLINK, etc)."""
```

Comment:
Nit: We shouldn't need to link to these, as the range of values is known from the type, and you can click and see the full list in the docs. I think either say `(file, directory, symlink, etc)` or drop them entirely (my slight preference).

Reply:
Sure, happy to remove the examples. I made this change because strings are now invalid arguments, and only enum members are expected, but it's very redundant with the typing.

---

While fixing this I noticed that the following attribute has:
```python
size: Optional[int]
"""Size of the file (will be 0 if ``type`` is not "file")."""
```
Which I've changed to:
```python
"""Size of the file (will be 0 if :attr:`type` is not :attr:`FileType.FILE`)."""
```
Let me know if you'd prefer something different here. I thought it was worth linking as `type` appears after `size` in the docs, since attributes are sorted alphabetically.

Reply:
Btw `size` is another case where `Optional` feels a little bit off to me. Not something to address now, but maybe I should add this as an example in that issue. But maybe `None` is actually correct here?

In pebble we have:
```go
Size         *int64   `json:"size,omitempty"`
```
Does this omit if size is 0?

In ops we have:
```python
_FileInfoDict.size: NotRequired[Optional[int]]  # could be NotRequired[int]
```
```python
# in FileInfo.from_dict
size=d.get('size')  # None if missing -- should it be 0 instead? Depends on pebble behaviour
```
```python
FileInfo.size: Optional[int]  # could remove Optional here as well if 0 is default value
"""Size of the file (will be 0 if ``type`` is not "file")."""
```

Reply:
Oh yes, it seems like the docstring is incorrect here. Pebble will omit it for non-files, so it'll come through as None. We should change the docstring to "will be None if ..." -- seems fine to do that in as a drive-by change.